kore/app/exec/Main.hs

```diff
@@ -402,16 +561,26 @@ main = do
 
 mainWithOptions :: KoreExecOptions -> IO ()
 mainWithOptions execOptions = do
-    let KoreExecOptions { koreLogOptions } = execOptions
-    exitCode <-
-        runKoreLog koreLogOptions
-        $ handle handleSomeException
-        $ handle handleSomeEntry
-        $ handle handleWithConfiguration go
-    let KoreExecOptions { rtsStatistics } = execOptions
-    Foldable.forM_ rtsStatistics $ \filePath ->
-        writeStats filePath =<< getStats
-    exitWith exitCode
+    let KoreExecOptions { koreLogOptions, bugReport } = execOptions
+    Temp.withSystemTempDirectory
+        (fromMaybe "report" $ toReport bugReport)
+        $ \tempDirectory -> do
+            traceM tempDirectory
+            exitCode <-
+                runKoreLog tempDirectory koreLogOptions
+                $ handle (handleSomeException tempDirectory)
+                $ handle handleSomeEntry
+                $ handle handleWithConfiguration go
+            let KoreExecOptions { rtsStatistics } = execOptions
+            Foldable.forM_ rtsStatistics $ \filePath ->
+                writeStats filePath =<< getStats
+            let reportPath = maybe tempDirectory ("./" <>) (toReport bugReport)
+            writeInReportDirectory tempDirectory
+                `catch` \(_ ::SomeException) ->
+                    archiveDirectoryReport tempDirectory reportPath
+            archiveDirectoryReport tempDirectory reportPath
```

Comment:
This will call `archiveDirectoryReport` twice if there is an error. Let's try this instead:

```suggestion
            finally
                (writeInReportDirectory tempDirectory)
                (archiveDirectoryReport tempDirectory reportPath)
```